unlock-app/src/components/creator/CreatorLockForm.js

```diff
@@ -37,15 +37,31 @@ export class CreatorLockForm extends React.Component {
     let expirationDuration = props.expirationDuration
     let keyPrice = props.keyPrice
     let maxNumberOfKeys = props.maxNumberOfKeys
-    if (props.convert) {
-      keyPrice = Web3Utils.fromWei(keyPrice, props.keyPriceCurrency)
-      expirationDuration = expirationDuration / props.expirationDurationUnit
-      // unlimited keys is represented differently in the frontend and the backend,
-      // so when a lock has `maxNumberOfKeys` set to UNLIMITED_KEYS_COUNT,
-      // convert it to infinity symbol for frontend display
-      if (maxNumberOfKeys === UNLIMITED_KEYS_COUNT) {
-        maxNumberOfKeys = INFINITY
+    try {
```

Comment:
Please add a comment to explain the rationale for the try catch thing. IIf I understanding things correctly this is because we invoke this with different formats which is probably something we should fix  (I k now I have an issue assigned to me for that!)

Reply:
👍